site/content/en/reference/fn/run/_index.md

```diff
@@ -38,99 +64,221 @@ kpt fn run DIR/ --fn-path FUNCTIONS_DIR/
 # functions may be scoped to a subset of Resources -- see `kpt help fn run`
 kpt fn run DIR/
 ```
+
 <!--mdtogo-->
 
-## Imperatively run a single function
+## Structured Results
 
-A function may be explicitly specified using the `--image` flag.
+Functions may emit results using the structure defined in the
+[typescript result] interface as an alternative to exiting with a non-zero
+status code. Users may want to store these results separately from
+configuration files. Kpt provides the `--results-dir` flag for users to
+specify a destination to write results to.
 
-__Example:__ Locally run the container image `gcr.io/example.com/my-fn` against
-the Resources in `DIR/`:
+**Example**: Run `validate-rolebinding` on an example package
 
 ```sh
-kpt fn run DIR/ --image gcr.io/example.com/my-fn
+kpt pkg get https://github.com/GoogleContainerTools/kpt-functions-sdk.git/example-configs example-configs
+mkdir results/
+kpt fn run example-configs/ --results-dir results/ --image gcr.io/kpt-functions/validate-rolebinding:results -- subject_name=bob@foo-corp.com
 ```
 
-If `DIR/` is not specified, the source will default to STDIN and sink will default
-to STDOUT.
+## Network Access
+
+By default, container functions cannot access network. `kpt` may enable network
+access using the `--network` flag, and specifying that a network is required in
+the functionConfig.
 
-__Example:__ this is equivalent to the preceding example:
+**Example**: Run `kubeval` on a package
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-config
+  annotations:
+    config.k8s.io/function: |
+      container:
+        image: gcr.io/kpt-functions/kubeval
+        network:
+          required: true
+    config.kubernetes.io/local-config: 'true'
+```
 
 ```sh
-kpt source DIR/ |
-kpt fn run --image gcr.io/example.com/my-fn |
-kpt sink DIR/
+kpt pkg get https://github.com/instrumenta/kubeval.git/fixtures .
+kpt fn source fixtures/*invalid.yaml |
+  kpt fn run --fn-path fc.yaml --network 2>error.txt || true
```

Comment:
so this requires both a flag here (imperatively) and in in the declared configuration? Is that by design?

How does the consumer know they have to provide the flag?

Also, can you provide an example that doesn't use source:

```kpt fn run PKG```

Reply:
In this case it's hard to change one thing at a time because deduplicating docs is pretty involved and each change can involve lots of files. I only changed 21 files, it's easier to review if you filter the files with .html or .xml extension using the file filter as those files are auto-generated. Another way is to review commit by commit as the "Docs: ..." commits are site changes and the "Regenerate docs" commits are auto-generated.

I will change this example.

Reply:
Let's close out this PR. I have made a note in Issue #907 to change this example and will follow up in a separate PR.

Reply:
I created a new issue to address the general case #933